### DISCUSSION TITLE
Run a job if fails

### DISCUSSION THREAD

**Discussion by gabsmprocha** [community/community#25307] [Actions]
<p>i had a workflow trigged by pull_request event, i have a job that would be executed only on failure of this event, but it doesn’t work… any ideas?</p>
<pre><code class="lang-auto">name: CI

on:
   pull_request:
        types: [opened, reopened, ready_for_review, edited, synchronize, closed]
        branches: [main]

....

 job-test:
        runs-on: ubuntu-latest
        if: ${{ github.event_name == 'pull_request' &amp;&amp; failure() }}
        steps:
         - name: testing
           run: echo "hello world"
</code></pre>

**Comment by Katsute** [comment]
<p>Since your <code>job-test</code> job doesn’t have a <code>needs</code>, it is possible that that job is running before any of your other jobs. You could add <code>needs</code> to that job, requiring all previous jobs to run first, before checking if there is a failure.</p>
<p><a href="https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds</a></p>

**Comment by vitalykarasik** [comment, marked as answer]
<p>Please see my job example below:</p>
<pre><code class="lang-auto">  delete-runner:
    needs: run-build
    runs-on: ubuntu-latest
    if: always() || cancelled()
</code></pre>
